vdev/src/commands/build.rs

```diff
@@ -0,0 +1,56 @@
+use clap::Args;
+
+use crate::app::Application;
+
+/// Build Vector
+#[derive(Args, Debug)]
+#[command()]
+pub struct Cli {
+    /// The build target e.g. x86_64-unknown-linux-musl
+    target: Option<String>,
+
+    /// Build with optimizations
+    #[arg(short, long)]
+    release: bool,
+
+    /// The feature to activate (multiple allowed)
+    #[arg(short = 'F', long)]
+    feature: Vec<String>,
+}
+
+impl Cli {
+    pub fn exec(&self, app: &Application) {
+        let mut command = app.command("cargo");
+        command.args(["build", "--no-default-features"]);
+
+        if self.release {
+            command.arg("--release");
+        }
+
+        command.arg("--features");
+        if !self.feature.is_empty() {
+            command.args([self.feature.join(",")]);
+        } else {
+            if app.platform.windows() {
+                command.arg("default-msvc");
+            } else {
+                command.arg("default");
+            }
+        };
```

Comment:
You may also find `script/features` interesting, to pull the required features out of a config file based on the components that are in use.

Reply:
Ah nice, I will incorporate that soon